### PR TITLE
organization billing

### PR DESCRIPTION
**Pull request by jjidt**

### CODE REVIEW

File: api/src/constants/billingPublicFields.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export const BILLING_PUBLIC_FIELDS = Object.freeze(['currency', 'name', 'balance'])
+
+export const BILLING_SOURCE_PUBLIC_FIELDS = Object.freeze([
+  'address_city',
+  'address_country',
+  'address_line1',
+  //address_line1_check,
```
Comment (dheniges) [inline api/src/constants/billingPublicFields.js:8]: are commented fields what's available to us? If so, maybe add a comment at the top of this constant?

File: api/src/controllers/orgBilling.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      resourceType: RESOURCE_TYPES.ORGANIZATION,
+      action: ACTIONS.UPDATE_MY_BILLING
+    }
+  })(req, res, next)
+}
+
+const orgBillingController = function(router) {
+  // all routes are nested at organizations/:orgId and receive req.params.orgId
+  router.get('/',
+    getOrgBilling
```
Comment (dheniges) [inline api/src/controllers/orgBilling.js:63]: There are no permissions on `getOrgBilling`?

File: api/src/coordinators/billing.js [hunk 3]
```diff
@@ -1,16 +1,59 @@
...
+  return formatBilling(customer)
+}
+
+const getOrgBilling = async function(orgId) {
+  const org = await BusinessOrganization.findById(orgId)
+
+  if (!org) throw ono({ code: 404 }, `Could not fetch billing, organization not found with id ${orgId}`)
+
+  // No billing information yet, return empty object
+  if (!org.stripeId) return null
```
Comment (dheniges) [inline api/src/coordinators/billing.js]: This is returning nothing in the body. Probably want an empty data object at minimum?

**Review by dheniges — CHANGES_REQUESTED** [review]
Overall looking good code-wise. But how do I manually test this to create the stripe billing stuffs?

**Comment by dheniges** [comment]
Also, when trying to update billing on the seed bonkey org I get this error:
```
danger-api | Error: No such token: tok_1ErpIbDBs6sqEFggtXQas4OV
danger-api |     at Constructor._Error (/api/node_modules/stripe/lib/Error.js:12:17)
danger-api |     at Constructor (/api/node_modules/stripe/lib/utils.js:134:13)
danger-api |     at new Constructor (/api/node_modules/stripe/lib/utils.js:134:13)
danger-api |     at Function.StripeError.generate (/api/node_modules/stripe/lib/Error.js:57:12)
danger-api |     at IncomingMessage.<anonymous> (/api/node_modules/stripe/lib/StripeResource.js:162:39)
danger-api |     at IncomingMessage.emit (events.js:194:15)
danger-api |     at IncomingMessage.EventEmitter.emit (domain.js:441:20)
danger-api |     at endReadableNT (_stream_readable.js:1103:12)
danger-api |     at process._tickCallback (internal/process/next_tick.js:63:19)
danger-api | PUT /api/organizations/100000/billing 500 288.228 ms - 1998
```

and FYI I reseeded dev to get the updated migrations

**Comment by jjidt** [comment]
I think the api .env stripe secret has to be from the same account as the publicly viewable key we're using client-side in the StripeContainer

try it again with my creds and see if it works, or grab the `pk_test` key for your account and replace it in StripeContainer

**Comment by jjidt** [comment]
@dheniges updated with a handful of changes

File: api/src/controllers/orgBilling.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+})
+
+const paramSchema = Joi.compile({
+  orgId: Joi.number().required()
+})
+
+const conditionalReadPerm = (req, res, next) => {
+  const { orgId } = req.params
+
+  console.log(orgId, req.requestorInfo.orgId)
```
Comment (dheniges) [inline api/src/controllers/orgBilling.js]: :eyes:

**Review by dheniges — CHANGES_REQUESTED** [review]
One sneaky console log. I was able to get it to work with your creds, looks good now 👍

File: api/src/controllers/orgBilling.js (hunk 4 above)
Comment (jjidt) [reply api/src/controllers/orgBilling.js]: oops, removed

**Review by dheniges — APPROVED** [review]

**Pull request merged**